tools/wpt/browser.py

```diff
@@ -758,3 +758,33 @@ def install_webdriver(self, dest=None, channel=None):
 
     def version(self, binary=None, webdriver_binary=None):
         return None
+
+
+class Epiphany(Browser):
+    """Epiphany-specific interface."""
+
+    product = "epiphany"
+    requirements = "requirements_epiphany.txt"
+
+    def install(self, dest=None, channel=None):
+        raise NotImplementedError
+
+    def find_binary(self, venv_path=None, channel=None):
+        return find_executable("epiphany")
+
+    def find_webdriver(self, channel=None):
+        return find_executable("WebKitWebDriver")
+
+    def install_webdriver(self, dest=None, channel=None):
+        raise NotImplementedError
+
+    def version(self, binary=None, webdriver_binary=None):
+        if binary is None:
+            return None
+        output = call(binary, "--version")
+        if output:
+            # Stable release output looks like: "Web 3.30.2"
+            # Tech Preview output looks like "Web 3.31.3-88-g97db4f40f"
+            # So split off the first word, then truncate after the first hyphen.
```

Comment:
Why not include the whole 3.31.3-88-g97db4f40f? Code that processes the version string (in wpt.fyi) could make it prettier if needed, but having the full string seems potentially useful if there can be updates which don't change the leading part. @lukebjerring opinions?

Reply:
Can keep it + adapt to this format - fyi @Hexcles - this may need to be handled in our version-string special cases.

Reply:
Is "Web" the name of the application? If so, the convention is to strip it, but all the remaining stuff can be kept, e.g. `3.30.2` or `3.31.3-88-g97db4f40f`.

Reply:
The codename of the project is "Epiphany" and that's probably what we'd want to display on wpt.fyi, but the actual user-visible name is just "Web". (We can also write "GNOME Web" when needed to disambiguate.)